packages/record/src/recorder.ts

```diff
@@ -133,28 +181,13 @@ export default class Recorder {
     this.jsPsych.pluginAPI.initializeCameraRecorder(stream, opts);
   }
 
-  /**
-   * Get recorder from jsPsych plugin API.
-   *
-   * If camera recorder hasn't been initialized, then return the microphone
-   * recorder.
-   *
-   * @returns MediaRecorder from the plugin API.
-   */
-  private get recorder() {
-    return (
-      this.jsPsych.pluginAPI.getCameraRecorder() ||
-      this.jsPsych.pluginAPI.getMicrophoneRecorder()
-    );
-  }
-
-  /**
-   * Get stream from either recorder.
-   *
-   * @returns MediaStream from the plugin API.
-   */
-  private get stream() {
-    return this.recorder?.stream;
+  /** Reset the recorder to be used again. */
+  public reset() {
+    if (this.stream.active) {
+      throw new StreamActiveOnResetError();
+    }
+    this.intializeRecorder(this.streamClone.clone());
+    this.blobs = [];
```

Comment:
Do we need to reset any of the states and properties back to initial values here, like `filename`, `micChecked`, `processorNode`, `s3`? 
And does it make sense clear the webcam feed here? (If so I can add it myself, just checking).

Reply:
I don't think so. 

- `filename`: Is changed on start.
- `S3`: Is created on start.
- `micChecked`: Does it need to be?  Once the mic has been checked, does it remain checked?
- `processorNode`: Also unsure.  What do you think?

Reply:
I'm also not sure. I'll figure it out when I pull your changes into my branch and let you know if I think anything needs to work differently.